### PR TITLE
Feat/shopping cart logic

### PR DESCRIPTION
**Pull request by Madlenne**

### CODE REVIEW

**Review by Pasik97 — APPROVED** [review]

**Review by Pasik97 — DISMISSED** [review]

File: frontend/src/domain/Catalogue/Catalogue.tsx [hunk 1]
```diff
@@ -57,15 +51,16 @@ export function Catalogue() {
...
             <GameCard
               key={index}
-              title={title}
-              categories={categories}
+              id={id}
+              title={product_name}
+              category={category}
               price={price}
-              cover={cover}
+              cover={witcher}
```
Comment (PiotrKolecki) [inline frontend/src/domain/Catalogue/Catalogue.tsx:63]: I this still a mock?

File: frontend/src/store/cart/selectors.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+export const getUserItems = (state: AppState, currentCustomerId: string | undefined ) => {
+
+  const cart = getCart(state);
+  const userItems = cart.filter(({ customer_id }) => customer_id === currentCustomerId) || [];
+  const userProductsId = userItems.flatMap(({ items }) => (items || []).map(({ product_id }) => product_id));
+
+  const userGames = getGameByIds(state, userProductsId);
+  let gamesWithQuantity: userGame[] = [];
+
+  userGames.forEach((game) => {
```
Comment (PiotrKolecki) [inline frontend/src/store/cart/selectors.ts:24]: This seams over complicated

**Review by PiotrKolecki — CHANGES_REQUESTED** [review]

File: frontend/src/store/cart/selectors.ts (hunk 2 above)
Comment (Madlenne) [reply frontend/src/store/cart/selectors.ts:24]: Agree but redux store structure turned out to be a little complicated. Everything is an array so we have to process the elements one by one in that way :(

File: frontend/src/domain/Catalogue/Catalogue.tsx (hunk 1 above)
Comment (Madlenne) [reply frontend/src/domain/Catalogue/Catalogue.tsx:63]: Yes, covers will be added in a separate PR

**Review by PiotrKolecki — APPROVED** [review]
Cover images will be added in different PR.

**Pull request merged**